src/pytest_plugins/skip_list/skip_list.py

```diff
@@ -0,0 +1,105 @@
+"""
+Pytest plugin for skipping tests based on a YAML skip list file.
+
+This plugin allows specifying tests to be skipped via:
+1. A YAML file containing node IDs (default: skip_tests.yaml)
+2. Command line argument --skip-list-file to specify custom file
+3. Command line argument --skip-tests for comma-separated patterns
+
+Follows patterns established by other pytest_plugins in this codebase.
+"""
+
+import fnmatch
+from pathlib import Path
+from typing import List
+
+import pytest
+import yaml
+
+
+def pytest_addoption(parser: pytest.Parser) -> None:
+    """Add command-line options for skip list functionality."""
+    skip_group = parser.getgroup("skip_list", "Skip tests based on configured patterns")
+
+    skip_group.addoption(
+        "--skip-list-file",
+        action="store",
+        dest="skip_list_file",
+        type=Path,
+        default="skip_tests.yaml",
+        help="Path to YAML file containing test node IDs to skip (default: skip_tests.yaml)",
+    )
+
+    skip_group.addoption(
+        "--skip-tests",
+        action="store",
+        dest="skip_tests_cli",
+        default="",
+        help="Comma-separated list of test node ID patterns to skip",
+    )
+
+
+@pytest.hookimpl(tryfirst=True)
+def pytest_configure(config: pytest.Config) -> None:
+    """Process skip list configuration and store patterns."""
+    skip_patterns: set[str] = set()
+
+    # Load from YAML file
+    skip_file = config.getoption("skip_list_file")
+    if skip_file and Path(skip_file).exists():
+        try:
+            with open(skip_file, "r") as f:
+                skip_data = yaml.safe_load(f) or {}
+
+            # Support both list format and dict format
+            if isinstance(skip_data, list):
+                skip_patterns.update(skip_data)
+            elif isinstance(skip_data, dict) and "skip_tests" in skip_data:
+                skip_list = skip_data["skip_tests"]
+                if isinstance(skip_list, list):
+                    skip_patterns.update(skip_list)
+
+        except (yaml.YAMLError, IOError) as e:
+            pytest.exit(f"Error reading skip list file {skip_file}: {e}")
+
+    # Load from command line
+    cli_patterns = config.getoption("skip_tests_cli")
+    if cli_patterns:
+        patterns = [p.strip() for p in cli_patterns.split(",") if p.strip()]
+        skip_patterns.update(patterns)
+
+    # Store patterns in config for use by collection hook
+    setattr(config, "_skip_list_patterns", skip_patterns)  # noqa: B010
+
+
+@pytest.hookimpl(tryfirst=True)
+def pytest_collection_modifyitems(config: pytest.Config, items: List[pytest.Item]) -> None:
+    """Skip items that match patterns in the skip list."""
+    skip_patterns: set[str] = getattr(config, "_skip_list_patterns", set())
+
+    if not skip_patterns:
+        return
+
+    skipped_items = []
```

Comment:
I'm not sure if this list is needed, since we are adding `skip` marker for the entries in the passed `items` in-place (and we don't use skipped_items directly).

As a side note, maybe it's worth logging the skipped elements